src/blockmap.jl

```diff
@@ -141,51 +149,58 @@ function Base.hvcat(rows::NTuple{nr,Int}, As::Union{LinearMap,UniformScaling}...
         end
     end
 
-    return BlockMap{T}(promote_to_lmaps(n, 1, As...), rows)
+    return BlockMap{T}(promote_to_lmaps(n, 1, 1, As...), rows)
```

Comment:
is the `if needcols` necessary? Is it not anyway worthwhile to check consistent total column size.

Also, since Github doesn't allow me to comment on old lines which have not been edited in this PR:
```julia
nci = nc ÷ rows[i]
nci * rows[i] != nc && throw(...)
```
can probably be captured by `nci, r = divrem(nc, rows[i]); r != 0 && throw(...)`

Reply:
I've copied this as is from Base.uniformscaling.jl. I think it might be necessary if you do have a row with only `UniformScaling`s. Honestly, I haven't even tried to parse the logic here, because I knew it was tested in Base, and does serve the exact same purpose as we want it here.

Regarding the old line, you should make a PR to Julia Base 😛. Anyway, I've included it in the PR.

Reply:
Judging from coverage test results, we do seem to have one test where the `needcols` branch is run. I strongly suspect it's this one:
```julia
@test [I; LA LA LA] isa LinearMaps.BlockMap{elty}
```
in line 541.

Reply:
I guess I misexpressed myself. The block inside the `if needcols` is definitely necessary. I was just wondering if we needed the `if` itself. This block can also be ran, and then it just checks whether total column size is consistent. If there is a row of `I`s, it will also compute their size.

Reply:
Cool. That turns out to be important. I had a test, which was supposed to be throwing, and it did throw, but not because of our code, but because of `LinearAlgebra`. Removing the `if`-branch, now, as you say, we check for consistent total column size, and the code throws as expected.